pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BrokerRequestHandler.java

```diff
@@ -304,6 +308,19 @@ public BrokerResponse processBrokerRequest(@Nonnull BrokerRequest brokerRequest,
   }
 
   /**
+   * Returns the time column name for the table name from the time boundary service.
+   * Can return null if the time boundary service does not have the information.
+   *
+   * @param tableName Name of table for which to get the time column name
+   * @return Time column name for the table.
+   */
+  @Nullable
+  private String getTimeColumnName(@Nonnull String tableName) {
+    TimeBoundaryInfo timeBoundary = _timeBoundaryService.getTimeBoundaryInfoFor(tableName);
+    return (timeBoundary != null) ? timeBoundary.getTimeColumn() : null;
```

Comment:
Not 100% sure, but IIRC there's no time boundary for offline-only tables nor for the realtime part of a hybrid table. In that case, that would mean that most of the time this optimization wouldn't apply.